client/client_update.go

```diff
@@ -191,7 +191,27 @@ func (u *UpdateClient) FetchUpdate(api ApiRequester, url string, maxWait time.Du
 
 type UpdateResponse struct {
 	*datastore.UpdateInfo
-	*updatecontrolmap.UpdateControlMap `json:"update_control_map"`
+
+	// Use a strict parser for the Update Control Map in order to reject
+	// unknown fields. This is in contrast to the rest of the response,
+	// where we allow unknown fields.
+	UpdateControlMap strictUpdateControlMapParser `json:"update_control_map"`
+}
+
+type strictUpdateControlMapParser struct {
+	*updatecontrolmap.UpdateControlMap
+}
+
+func (s *strictUpdateControlMapParser) UnmarshalJSON(data []byte) error {
```

Comment:
I wonder if we could now reuse this `strictUpdateControlMapParser` struct in D-Bus Update Control Map parsing [here](https://github.com/mendersoftware/mender/blob/master/app/updatemanager.go#L114-L121).

I think that reusing the struct we have less chances for them to diverge in the future (as it happened now, right? Because D-Bus parsing was disallowing extra fields, but here we were not)

Reply:
Good idea, I'll try to unify them.

Reply:
Yes, totally agree! But should we then add it to the `UpdateControl` type?

Reply:
Yes! I will try this. Maybe we can even get rid of `Validate` as a separate step! :smiley:

Reply:
This actually required quite a big change because the struct needs to be initialized differently, so I added it as a separate commit. I recommend viewing that one in particular with whitespace turned off.